cmd/cli/docker/docker_run.go

```diff
@@ -139,101 +134,127 @@ func dockerRun(cmd *cobra.Command, cmdArgs []string, opts *DockerRunOptions) err
+		if !containsTag {
+			cmd.PrintErrln("Using default tag: latest. Please specify a tag/digest for better reproducibility.")
+		}
+	}
+
+	api := util.GetAPIClientV2(cmd)
+	resp, err := api.Jobs().Put(ctx, &apimodels.PutJobRequest{Job: job})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to submit job: %w", err)
 	}
 
-	return printer.PrintJobExecutionLegacy(ctx, executingJob, cmd, opts.DownloadSettings, opts.RunTimeSettings, util.GetAPIClient(ctx))
+	if len(resp.Warnings) > 0 {
+		printWarnings(cmd, resp.Warnings)
+	}
+
+	if err := printer.PrintJobExecution(ctx, resp.JobID, cmd, &opts.RunTimeSettings.RunTimeSettings, api); err != nil {
+		return fmt.Errorf("failed to print job execution: %w", err)
+	}
+
+	return nil
 }
 
-// CreateJob creates a job object from the given command line arguments and options.
-func CreateJob(ctx context.Context, image string, parameters []string, opts *DockerRunOptions) (*model.Job, error) {
-	outputs, err := parse.JobOutputs(ctx, opts.SpecSettings.OutputVolumes)
-	if err != nil {
-		return nil, err
+// TODO(forrest) [refactor]: dedupe from wasm_run
+func printWarnings(cmd *cobra.Command, warnings []string) {
+	cmd.Println("Warnings:")
+	for _, warning := range warnings {
+		cmd.Printf("\t* %s\n", warning)
 	}
+}
 
-	nodeSelectorRequirements, err := parse.NodeSelector(opts.SpecSettings.Selector)
+func CreateJobDocker(image string, parameters []string, opts *DockerRunOptions) (*models.Job, error) {
+	engineSpec, err := models.DockerSpecBuilder(image).
+		WithParameters(parameters...).
+		WithWorkingDirectory(opts.WorkingDirectory).
+		WithEntrypoint(opts.Entrypoint...).
+		WithEnvironmentVariables(opts.SpecSettings.EnvVar...).Build()
 	if err != nil {
 		return nil, err
 	}
 
-	labels, err := parse.Labels(ctx, opts.SpecSettings.Labels)
-	if err != nil {
-		return nil, err
+	// TODO(forrest) [refactor]: this logic is duplicated in wasm_run
+	resultPaths := make([]*models.ResultPath, 0, len(opts.SpecSettings.OutputVolumes))
+	for name, path := range opts.SpecSettings.OutputVolumes {
+		resultPaths = append(resultPaths, &models.ResultPath{
+			Name: name,
+			Path: path,
+		})
 	}
 
-	spec, err := legacy_job.MakeDockerSpec(
-		image, opts.WorkingDirectory, opts.Entrypoint, opts.SpecSettings.EnvVar, parameters,
-		legacy_job.WithResources(
-			opts.ResourceSettings.CPU,
-			opts.ResourceSettings.Memory,
-			opts.ResourceSettings.Disk,
-			opts.ResourceSettings.GPU,
-		),
-		legacy_job.WithNetwork(
-			opts.NetworkingSettings.Network,
-			opts.NetworkingSettings.Domains,
-		),
-		legacy_job.WithTimeout(opts.SpecSettings.Timeout),
-		legacy_job.WithInputs(opts.SpecSettings.Inputs.Values()...),
-		legacy_job.WithOutputs(outputs...),
-		legacy_job.WithAnnotations(labels...),
-		legacy_job.WithNodeSelector(nodeSelectorRequirements),
-		legacy_job.WithDeal(
-			opts.DealSettings.TargetingMode,
-			opts.DealSettings.Concurrency,
-		),
-	)
+	task, err := models.NewTaskBuilder().
+		Name("TODO").
```

Comment:
We currently don't have a means for users to specify a task name with the current flags on `wasm run` or `docker run` I could stub in a UUID here and add a `--task-name` flag later, thoughts?

Reply:
we can use "main" as the default task name. Task names are only useful when we support multiple tasks per job, and that won't be doable through `docker run` and `wasm run` as they will most likely submit a single task per job. So calling it "main" or any other standard name is acceptable